Problem with WithProxy method

use code：
```go
	ne, _ := nuclei.NewNucleiEngine(nuclei.WithTemplatesOrWorkflows(nuclei.TemplateSources{Templates: []string{yourtemplates}}),
		nuclei.WithGlobalRateLimit(150, time.Second),
		nuclei.WithNetworkConfig(nuclei.NetworkConfig{Timeout: 10}),
		nuclei.WithProxy([]string{"socks5://127.0.0.1:7890"}, false),
		//nuclei.WithVerbosity(nuclei.VerbosityOptions{Debug: true}),
	)
```
	
go panic：

```go                                                                                                                          
goroutine 1 [running]:                                                                                                                   
github.com/projectdiscovery/utils/maps.(*SyncLockMap[...]).Get(0xc0010260c8?, {0xc000c0ac20?, 0x10?})                                    
        C:/Users//go/pkg/mod/github.com/projectdiscovery/utils@v0.0.92/maps/synclock_map.go:71 +0x36                               
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/httpclientpool.wrappedGet(0xc000781808, 0xc001027a80)                           
        C:/Users//go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.7/pkg/protocols/http/httpclientpool/clientpool.go:175 +0x96
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/httpclientpool.Get(0xc000781808?, 0xc000c1a100?)                                
        C:/Users//go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.7/pkg/protocols/http/httpclientpool/clientpool.go:163 +0x4b
github.com/projectdiscovery/nuclei/v3/lib.(*NucleiEngine).init(0xc0013b0e08)                                                             
        C:/Users//go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.7/lib/sdk_private.go:112 +0xec                             
github.com/projectdiscovery/nuclei/v3/lib.NewNucleiEngine({0xc001027ec0, 0x4, 0xba79e5?})                                                
        C:/Users//go/pkg/mod/github.com/projectdiscovery/nuclei/v3@v3.2.7/lib/sdk.go:276 +0x158                                    
main.ForKeyNucleiRun({0xc00004e240, 0x1a}, {0xc00000a0e8, 0x5}, {0xc00005a0b8?, 0x4829920?})                                             
        C:/Users//Desktop/nucleiv3/main.go:265 +0x2c7                                                                              
main.main()                                                                                                                              
        C:/Users//Desktop/nucleiv3/main.go:144 +0x46

```

@dogancanbakir

`	ne, err := nuclei.NewNucleiEngineCtx(context.Background(),
		nuclei.DisableUpdateCheck(), // 禁用nuclei插件自动更新
		nuclei.WithTemplateFilters(nuclei.TemplateFilters{ExcludeSeverities: "info"}), // 取消扫描info类插件
		nuclei.WithTemplatesOrWorkflows(nuclei.TemplateSources{Templates: []string{"D:\\tools\\nuclei-templates"}}),
		nuclei.WithProxy([]string{"socks5://192.168.2.245:8080"}, false),
	)`



panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x1 addr=0x14 pc=0x2ab70b6]

goroutine 1 [running]:
github.com/projectdiscovery/utils/maps.(*SyncLockMap[...]).Get(0xc000df5290?, {0xc000050480?, 0x0?})
	D:/gopyoject/fastscan/vendor/github.com/projectdiscovery/utils/maps/synclock_map.go:71 +0x36
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/httpclientpool.wrappedGet(0xc00049c808, 0xc00103e998)
	D:/gopyoject/fastscan/vendor/github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/httpclientpool/clientpool.go:179 +0x65
github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/httpclientpool.Get(0xc00049c808?, 0xc00103db40?)
	D:/gopyoject/fastscan/vendor/github.com/projectdiscovery/nuclei/v3/pkg/protocols/http/httpclientpool/clientpool.go:171 +0x52
github.com/projectdiscovery/nuclei/v3/lib.(*NucleiEngine).init(0xc0003a5188, {0x413c158, 0x5c72320})
	D:/gopyoject/fastscan/vendor/github.com/projectdiscovery/nuclei/v3/lib/sdk_private.go:112 +0x105
github.com/projectdiscovery/nuclei/v3/lib.NewNucleiEngineCtx({0x413c158, 0x5c72320}, {0xc00103f760, 0x4, 0x1?})
	D:/gopyoject/fastscan/vendor/github.com/projectdiscovery/nuclei/v3/lib/sdk.go:285 +0x15a

--- a/lib/sdk_private.go
+++ b/lib/sdk_private.go
@@ -108,14 +108,6 @@ func (e *NucleiEngine) init(ctx context.Context) error {
 		return err
 	}
 
-	if e.opts.ProxyInternal && types.ProxyURL != "" || types.ProxySocksURL != "" {
-		httpclient, err := httpclientpool.Get(e.opts, &httpclientpool.Configuration{})
-		if err != nil {
-			return err
-		}
-		e.httpClient = httpclient
-	}
-
 	e.parser = templates.NewParser()
 
 	if sharedInit == nil || protocolstate.ShouldInit() {
@@ -125,6 +117,14 @@ func (e *NucleiEngine) init(ctx context.Context) error {
 	sharedInit.Do(func() {
 		_ = protocolinit.Init(e.opts)
 	})
+
+	if e.opts.ProxyInternal && types.ProxyURL != "" || types.ProxySocksURL != "" {
+		httpclient, err := httpclientpool.Get(e.opts, &httpclientpool.Configuration{})
+		if err != nil {
+			return err
+		}
+		e.httpClient = httpclient
+	}
 
 	e.applyRequiredDefaults(ctx)
 	var err error

